### PR TITLE
python-pycares: add new package

### PR DESCRIPTION
**Pull request by dynasticorpheus**
Maintainer: @dynasticorpheus 

Compile / Run tested: Linksys_wrt32x, mvebu (cortex-a9), OpenWrt 22.03.0-rc6

Description: Add new package [pycares](https://pypi.org/project/pycares/)

### CODE REVIEW

**Comment by dynasticorpheus** [comment]
cc python maintainers: @commodo , @jefferyto

File: lang/python/python-pycares/Makefile
Comment (commodo) [inline lang/python/python-pycares/Makefile:32]: the `define Package/python3-pycares-src` definition is redundant;

you get it for free when you define `Package/python3-pycares`

you can remove it;

File: lang/python/python-pycares/Makefile
Comment (commodo) [inline lang/python/python-pycares/Makefile:48]: since i'm complaining anyway: you can remove this empty last-line; but it's more of a nitpick (or personal feeling)

feel free to disregard it;

File: lang/python/python-pycares/Makefile
Comment (commodo) [inline lang/python/python-pycares/Makefile:6]: depends on who you ask, you can also just do `PKG_RELEASE:=$(AUTORELEASE)`

i'm not sure how robust the mechanism is, but i use it;
it makes my life simple when updating a package, as there is one less line to edit; 

otherwise, the rule is: when updating to a new version, PKG_RELEASE gets reset to 1; otherwise, if something gets changed/updated, you need to bump PKG_RELEASE, which (as it sounds) is a pretty manual and (from experience) a bit of an error-prone process;

**Comment by commodo** [comment]
rest looks good to me;

i'm assuming you tested with just `python3-light`

File: lang/python/python-pycares/Makefile
Comment (dynasticorpheus) [reply lang/python/python-pycares/Makefile]: Thanks, I'm happy to follow your lead here so updated accordingly next to the other suggestions

**Review by commodo — APPROVED** [review]

**Comment by commodo** [comment]
will merge this in 1-2 days if no objection

**Comment by dynasticorpheus** [comment]
> will merge this in 1-2 days if no objection

Sure no problem, did add python3-cffi dependency just now based on additional bare install testing. Apologies for the force-push mess.

**Comment by commodo** [comment]
Force-pushing is fine.

Since you're doing it, might as well just squash commits into a single one :)

**Comment by dynasticorpheus** [comment]
> Force-pushing is fine.
> 
> Since you're doing it, might as well just squash commits into a single one :)

I learned something today! :)

**Comment by commodo** [comment]
> > Force-pushing is fine.
> > Since you're doing it, might as well just squash commits into a single one :)
> 
> I learned something today! :)

well, (often in various projects) the idea is that fix-commits are not needed until they are merged;
once something is in master (or merged), only then fix-commits are more acceptable;
up until then, it's WIP

this is because (over time), nobody will care (when looking into the git history) about stuff that was fixed before it was in master;
also, i only search for fix-commits (in git history), if i find something broken (that has appeared in master or a release)

but, each repo/project has different rules

**Pull request merged**
